service/src/main/java/com/commercetools/pspadapter/payone/transaction/BaseDefaultTransactionExecutor.java

```diff
@@ -0,0 +1,169 @@
+package com.commercetools.pspadapter.payone.transaction;
+
+import com.commercetools.pspadapter.payone.domain.ctp.CustomTypeBuilder;
+import com.commercetools.pspadapter.payone.domain.ctp.PaymentWithCartLike;
+import com.commercetools.pspadapter.payone.domain.payone.PayonePostService;
+import com.commercetools.pspadapter.payone.domain.payone.exceptions.PayoneException;
+import com.commercetools.pspadapter.payone.domain.payone.model.common.BaseRequest;
+import com.commercetools.pspadapter.payone.domain.payone.model.common.PayoneResponseFields;
+import com.commercetools.pspadapter.payone.domain.payone.model.common.ResponseStatus;
+import com.commercetools.pspadapter.payone.mapping.CustomFieldKeys;
+import com.commercetools.pspadapter.payone.mapping.PayoneRequestFactory;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.sphere.sdk.client.BlockingSphereClient;
+import io.sphere.sdk.payments.Payment;
+import io.sphere.sdk.payments.Transaction;
+import io.sphere.sdk.payments.TransactionState;
+import io.sphere.sdk.payments.commands.PaymentUpdateCommand;
+import io.sphere.sdk.payments.commands.updateactions.*;
+import io.sphere.sdk.types.CustomFields;
+import io.sphere.sdk.types.Type;
+import org.slf4j.Logger;
+
+import javax.annotation.Nonnull;
+import java.time.ZonedDateTime;
+import java.util.*;
+
+
+/**
+ * Base class for validating/executing/re-trying preauthorization/authorization (authorization/charge) transactions.
+ */
+abstract public class BaseDefaultTransactionExecutor extends TransactionBaseExecutor {
+
+    protected PayoneRequestFactory requestFactory;
+    protected final PayonePostService payonePostService;
+
+    public BaseDefaultTransactionExecutor(@Nonnull LoadingCache<String, Type> typeCache,
+                                          @Nonnull final PayoneRequestFactory requestFactory,
+                                          @Nonnull final PayonePostService payonePostService,
+                                          @Nonnull BlockingSphereClient client) {
```

Comment:
do you have to use the Blockingsphere client here?

Reply:
Old legacy issue, not easy to fix, but separate task is create for the next milestone: https://github.com/commercetools/commercetools-payone-integration/issues/203